src/Scanner/Scanner.ts

```diff
@@ -18,8 +18,17 @@ export default class {
   cacheScanner: IntervalId;
   chainScanner: IntervalId;
 
-  buckets: IBuckets;
-  eventWatchers: {};
+  buckets: IBuckets = {
+    currentBuckets: {
+      blockBucket: -1,
+      timestampBucket: -1,
+    },
+    nextBuckets: {
+      blockBucket: -1,
+      timestampBucket: -1,
+    }
+  };
+  eventWatchers: {} = {};
```

Comment:
Doesn't `eventWatchers: {}` mean that the only valid value is an empty object?  Are we sure we would only allow an empty object there?
https://stackoverflow.com/questions/36967176/what-is-type